app/components/CrashRecommendationCard.tsx

```diff
@@ -0,0 +1,297 @@
+import { useState, useCallback } from "react";
+import Button from "react-bootstrap/Button";
+import Card from "react-bootstrap/Card";
+import Form from "react-bootstrap/Form";
+import Spinner from "react-bootstrap/Spinner";
+import { useForm } from "react-hook-form";
+import { useQuery, useMutation } from "@/utils/graphql";
+import CrashRecommendationParters from "./CrashRecommendationPartners";
+import {
+  RECOMMENDATION_STATUS_QUERY,
+  RECOMMENDATION_PARTNERS_QUERY,
+  INSERT_RECOMMENDATION_MUTATION,
+  UPDATE_RECOMMENDATION_MUTATION,
+} from "@/queries/recommendations";
+import {
+  Recommendation,
+  RecommendationStatus,
+  RecommendationPartner,
+  CoordinationPartner,
+  RecommendationFormInputs,
+} from "@/types/recommendation";
+import { useAuth0 } from "@auth0/auth0-react";
+
+/**
+ * Compares the old vs new RecommendationPartner arrays and returns
+ * new arrays of partner data that can be used in graphql mutation
+ * @param recommendationId - the pk of the current recommendation, if we are editing
+ * @param partnersOld - array of RecommendationPartner[] objs
+ * @param partnersNew - array of RecommendationPartner's from the form data. it is
+ * expected that these objects may only have a `partner_id` property
+ * @returns [addPartners, deletePartnerPks]
+ */
+const getPartnerChanges = (
+  recommendationId: number | null,
+  partnersOld: Partial<RecommendationPartner>[],
+  partnersNew: Partial<RecommendationPartner>[]
+) => {
+  const addPartners: Partial<RecommendationPartner>[] = [];
+  const deletePartnerPks: number[] = [];
```

Comment:
these feel like function names, i think `partnersToAdd` and `partnersToDelete` or something like that would be more obvious that these are just arrays containing the partners to add or delete